### ISSUE TITLE
win10 alt + key for menu bar not working when scroll lock is on

### ISSUE THREAD

**Issue by binbinfrog** [Microsoft/vscode#19011]
- VSCode Version: 1.8.1
- OS Version: win 10

Steps to Reproduce:

1. open a vscode window

**2. toggle scroll lock on**
3. press alt then any other menu bar shortcut key to toggle a menu but failed

3. press alt + (any menu bar shortcut key) the menu won't show

won't reproduce on fedora os because the scroll lock seems to be disabled ^_^

I don't think it is a big problem, I don't use scroll lock at all, I just think it's abnormal

**Comment by monk-time** [comment]
Had the same bug on Win7/1.16.0 today. Very confusing because it's very easy to hit Scroll Lock by accident and not notice that it's enabled (my laptop doesn't even have an indicator light for it).

**Comment by bpasero** [comment]
Still happening?

**Comment by monk-time** [comment]
@bpasero, yes, still happening on 1.18. It's trivially easy to reproduce, Scroll Lock always breaks Alt+F/E/etc.

**Comment by bpasero** [comment]
@monk-time also with other apps or just VS Code?

**Comment by binbinfrog** [comment]
@bpasero 
at least not on windows 10 file explorer

**Comment by monk-time** [comment]
@bpasero Only VSCode. That's the first program ever to have its shortcuts be affected by Scroll Lock, and I am sure it is caused by a non-native Electon menu not following a multitude of Windows menu conventions (there are many tickets here about this). I have just checked 10 or so well-established Windows programs that I have running, none of them are affected.

**Comment by Hectate** [comment]
Just wanted to speak up and say that I encountered this issue today. To make it worse, I was working on a Dell laptop that does not have a scroll lock indicator or dedicated button. With the numerous other menu-related issues (e.g. Electron https://github.com/electron/electron/issues/2504) I went down a few rabbit holes before finding that scroll-lock would change this.

To any future people; it may be possible to switch Scroll-lock by pressing Fn+S (which I had done so inadvertently) if this is causing you issues.

**Comment by bpasero** [comment]
@Hectate @binbinfrog @monk-time can you try to reproduce this with our new custom menu, just configure `"window.titleBarStyle": "custom"`

**Comment by monk-time** [comment]
@bpasero Scroll lock causes no issues for me on the latest release with the custom menu enabled. :+1: 

I've noticed though that there are a couple of new minor convention-breaking issues with it though: 
* after opening a menu (e.g. `Alt+F`) I can't close it and return focus to where it was before by tapping `Alt` (a nice Windows convention), only by hitting `Esc` twice (a bit more awkward). 

  Examples of programs that close an expanded menubar and return focus on `Alt`: Windows Explorer (W7), Firefox, Adobe Reader, N++, foobar2000, qBittorrent.

* you can't re-use menubar shortcuts (`Alt+F`, `Alt+E` etc.) for other actions, both actions fire at once and the focus stays on the menu. I know it's usually not recommended to shadow those shortcuts, but they are very handy, so it would be great if the user-defined shortcuts overrode the menu completely.

Tested on 1.26.0. I haven't checked to see if there are tickets already for these issues, sadly I don't have the time atm to open them myself.

**Comment by bpasero** [comment]
We will change to custom menus on Windows. For now you can set `"window.titleBarStyle": "custom"`.

Please report individual issues for things you see with the custom menu.

**Comment by monk-time** [comment]
@bpasero Thanks, I've reported the first item (#56652) and found an existing issue for the second (#53654).
